### DISCUSSION TITLE
sidebar/sanfene/jvm

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#85] [Announcements]
# sidebar/sanfene/jvm

图文详解 50 道Java虚拟机高频面试题，这次面试，一定吊打面试官

https://tobebetterjavaer.com/sidebar/sanfene/jvm.html

**Comment by liujcstar** [comment]
图全没了,大佬

**Comment by Dabao-Su** [comment]
那啥，JDK8中常量池和静态变量还是在堆里面的，内存管理第3点你写错了

**Reply by Dabao-Su** [reply]
应该是说字符串常量池和静态变量还在堆里面，运行时常量池不管是JDK6、7、8都还是在方法区，而且这只是 HotSpot 虚拟机的实现

**Reply by createryu** [reply]
其实在堆里的含义是方法区也属于堆里啊

**Comment by Kddk-K** [comment]
24.10.8一刷

**Comment by hachinekooo** [comment]
- 20241214第一刷

**Reply by hachinekooo** [reply]
2025.1.20

**Comment by Yvonne-bit-stack** [comment]
2024.12.25

**Reply by hachinekooo** [reply]
2024.12.26

**Comment by gfvmjcdfg** [comment]
元空间是放在本地内存啊，直接内存不是用来NIO的吗？

**Comment by YaryumDou** [comment]
53实现热部署功能那里，为什么是重写findClass(),不是要重写loadClass()才能打破授权委派模型吗

**Comment by fantasy-zrq** [comment]
12.Java 堆的内存分区了解吗？
对于这个第12部分原文中的表述是：”当老年代内存不足时，会触发 Major GC，对整个堆进行垃圾回收。“
这个表述是有问题的，这里触发Major GC只是回收老年代中的对象，不会回收整个堆中的对象，回收整个堆中的对象是Full GC。

**Comment by student599** [comment]
GC的标记应该是标记不会被回收的对象吧
